### PR TITLE
serial_terminal: die in case unsuccessful login

### PR DESCRIPTION
**Pull request by asmorodskyi**
die in case unsuccessful login

VR: http://g22.suse.de/tests/68

### CODE REVIEW

File: lib/serial_terminal.pm [hunk 1]
```diff
@@ -84,12 +84,14 @@ sub login {
...
     type_string("$user\n");
-    wait_serial(qr/Password:\s*$/i);
-    type_password;
-    type_string("\n");
-    wait_serial(qr/$escseq* \w+:~\s\# $escseq* \s*$/x);
+    if (length $testapi::password) {
+        die 'Failed to wait for password prompt' unless wait_serial(qr/Password:\s*$/i);
+        type_password;
+        type_string("\n");
+        die 'Failed to confirm that login was successful' unless wait_serial(qr/$escseq* \w+:~\s\# $escseq* \s*$/x);
```
Comment (Vogtinator) [inline lib/serial_terminal.pm]: Does this also match for `$user` != root?

Comment (asmorodskyi) [reply lib/serial_terminal.pm]: No idea , do you know scenario where password-less non-root user trying to login ? I will clone it and check .

Comment (Vogtinator) [reply lib/serial_terminal.pm]: That would only apply to non-root with password

Comment (asmorodskyi) [reply lib/serial_terminal.pm]: the trick that no one actually using serial terminal for non-root users , I have checked our codebase there is not a single call for `select_serial_terminal(0);`

Comment (Vogtinator) [reply lib/serial_terminal.pm]: In that case it should be fine. It means that the $user parameter is pretty broken at the moment though.

Comment (asmorodskyi) [reply lib/serial_terminal.pm]: it is not "broken" . we just have some amount of dead code . some logic in `select_serial_terminal` which decided is it root or not . we even have function `select_user_serial_terminal` which is no one calling ...

**Pull request merged**